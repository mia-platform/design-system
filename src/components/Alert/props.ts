/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { IconComponent } from '../Icon/Icon.props.ts'
import { Type } from './types.ts'

export type AlertProps = {

  /**
   * The title displayed at the top of the alert.
   */
  title?: ReactNode

  /**
   * The description providing additional details about the alert.
   */
  description?: ReactNode

  /**
   * Custom children content to be rendered inside the alert.
   */
  children?: ReactNode

  /**
   * Indicates if the alert is closable.
   */
  isClosable?: boolean

  /**
   * The type of alert.
   */
  type?: Type

  /**
   * If true, renders a compact version of the alert with reduced padding.
   */
  isCompressed?: boolean

  /**
   * A custom icon to display in the alert. Replaces the default icon based on the `type`.
   */
  icon?: IconComponent

  /**
   * A custom action element (e.g., a button) to display inside the alert.
   */
  action?: ReactNode

  /**
   * A callback function triggered when the alert is closed.
   */
  onClose?: () => void
}
