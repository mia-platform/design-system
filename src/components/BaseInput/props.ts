import { InputRef } from 'antd'
import { Ref } from 'react'

import { Appearance } from './types.ts'

export type BaseInputProps = {

  /**
   * The placeholder for the input.
   */
  placeholder?: string

  /**
   * The type of the Input.
   */
  appearance?: Appearance

  /**
   * Whether the input is disabled.
   */
  isDisabled?: boolean,

  /**
   * Whether the input has error.
   */
  isError?: boolean

  /**
   * Whether the input is read-only.
   */
  isReadOnly?: boolean

  /**
   * Whether the input is full width. Defaults to true.
   */
  isFullWidth?: boolean

  /**
   * Ref to the antd input.
   */
  inputRef?: Ref<InputRef>

  /**
   * Classes for the input.
   */
  className?: string

}
