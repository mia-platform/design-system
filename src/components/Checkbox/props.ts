/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { CheckboxChangeEvent } from 'antd/es/checkbox'
import { ReactNode } from 'react'

export type CheckboxProps = {

  /**
   * Additional description of the checkbox, to be rendered below the checkbox `text`.
   * Is ignored if the `text` prop is not provided.
   */
  description?: ReactNode,

  /**
   * Allows you to control whether the checkbox is checked.
   */
  isChecked?: boolean,

  /**
   * Allows you to control whether the checkbox is disabled. Defaults to false.
   */
  isDisabled?: boolean,

  /**
   * Allows you to control whether the checkbox is checked on its first render. Defaults to false.
   */
  isInitiallyChecked?: boolean,

  /**
   * Allows you to control whether the checkbox is indeterminate. Defaults to false.
   */
  isIndeterminate?: boolean

  /**
   * Function that is invoked when the checkbox state is changed.
   */
  onChange?: (event: CheckboxChangeEvent) => void

  /**
   * Text to be displayed next to the checkbox.
   */
  text?: ReactNode
}
