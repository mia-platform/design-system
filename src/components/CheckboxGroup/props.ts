/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ReactNode } from 'react'

import { Direction } from './types.ts'

export type CheckboxGroupOption<T> = {

  /**
   * Adds the Radio id attribute value
   */
  value: T;

  /**
   * The text used to display the radio option
   */
  label: ReactNode;

  /**
   * description of the label that is placed underneath the label
   */
  description?: ReactNode;

  /**
   * Specifies whether the Radio option is disabled
   */
  disabled?: boolean;

}

export type CheckboxGroupProps<T> = {

  /**
   * The selected value displayed on first render
   */
  defaultValue?: T[];

  /**
   * Used for setting the currently selected value
   */
  value?: T[];

  /**
   * Options of the checkbox group
   */
  options: CheckboxGroupOption<T>[];

  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (checkedValue: T[]) => void;

  /**
   * If true, disable all checkboxes
   */
  isDisabled?: boolean;

  /**
   * The direction of the radio group. Defaults to 'vertical'
   */
  direction?: Direction
}
