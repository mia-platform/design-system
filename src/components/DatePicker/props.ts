/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  NoUndefinedRangeValueType as AntNoUndefinedRangeValueType,
  RangeValueType as AntRangeValueType,
} from 'rc-picker/lib/PickerInput/RangePicker'
import {
  RangeTimeProps as AntRangeTimeProps,
} from 'rc-picker/lib/interface'
import { Dayjs } from 'dayjs'

import { ShowTimeOptions } from './types'

type CommonProps = {
  canClear?: boolean
  format?: string
  minDate?: Dayjs
  maxDate?: Dayjs
  showTime?: boolean | ShowTimeOptions
  isErrorStatus?: boolean
  hasNowButton?: boolean
}

export type DatePickerProps = CommonProps & {
  onChange?: (date: Dayjs | Dayjs[], dateString: string | string[]) => void
  placeholder?: string
  defaultValue?: Dayjs
  value?: Dayjs
  isDisabled?: boolean
}

export type NoUndefinedRangeValueType = AntNoUndefinedRangeValueType<Dayjs>
export type RangeValueType = AntRangeValueType<Dayjs>
export type RangeTimeProps = AntRangeTimeProps<Dayjs>

export type RangePickerProps = CommonProps & {
  canBeEmpty?: boolean
  format?: string
  onChange?: (dates: NoUndefinedRangeValueType | null, dateStrings: [string, string]) => void
  placeholder?: [string, string]
  showTime?: boolean | ShowTimeOptions
  value?: RangeValueType
  defaultValue?: RangeValueType
  isDisabled?: boolean | [boolean, boolean]
}
