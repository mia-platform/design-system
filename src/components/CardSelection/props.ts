/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { InputType, Layout } from './types.ts'
import { IconComponent } from '../Icon/Icon.props.ts'

export type CardSelectionOption<T> = {

  /**
   * Optional id value for the card.
   */
  id?: string

  /**
   * The main title displayed prominently on the card.
   */
  title: ReactNode

  /**
   * An optional icon displayed alongside the title.
   */
  icon?: IconComponent

  /**
   * A subtitle displayed below the title.
   */
  subtitle?: ReactNode

  /**
   * Custom content to be rendered inside the card.
   */
  content?: ReactNode

  /**
   * The value associated with the card.
   */
  value?: T

  /**
   * Determines if the card is disabled.
   */
  disabled?: boolean
}

export type CardSelectionProps<T> = {

  /**
   * The default value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  defaultValue?: T | T[]

  gap?: number

  /**
   * The inputType of the card selection. Defaults: undefined (no input will be displayed).
   */
  inputType?: InputType,

  isDisabled?: boolean

  /**
   * The layout of the cards. Can be "horizontal" or "vertical". Default: "vertical".
   */
  layout: Layout

  /*
  * Handler triggered when the value of the selection changes (only triggered if inputType is "checkbox" or "radio")
  * */
  onChange?: (value: T | T[]) => void

  /*
  * Handler triggered when a card is clicked
  * */
  onClick?: (value?: T) => void

  /**
   * The options array for the card selection.
   */
  options: CardSelectionOption<T>[]

  /**
   * The controlled value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  value?: T | T[]
}
