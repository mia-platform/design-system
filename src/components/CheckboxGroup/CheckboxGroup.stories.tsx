/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Meta, StoryObj } from '@storybook/react'

import { CheckboxGroupOption, CheckboxGroupProps } from './props'
import { CheckboxGroup } from './CheckboxGroup'

const baseOptions: CheckboxGroupOption<number>[] = [
  {
    label: 'Option 1',
    value: 1,
  },
  {
    label: 'Option 2',
    value: 2,
  },
]

const optionsWithDescription: CheckboxGroupOption<number>[] = [
  {
    label: 'Option 1',
    value: 1,
    description: 'Description 1',
  },
  {
    label: 'Option 2',
    value: 2,
    description: 'Description 2',
  },
]

const optionsPartiallyDisabled: CheckboxGroupOption<number>[] = [
  {
    label: 'Option 1',
    value: 1,
    description: 'Description 1',
    disabled: true,
  },
  {
    label: 'Option 2',
    value: 2,
    description: 'Description 2',
  },
]

const allOptionsDisabled: CheckboxGroupOption<number>[] = [
  {
    label: 'Option 1',
    value: 1,
    description: 'Description 1',
    disabled: true,
  },
  {
    label: 'Option 2',
    value: 2,
    description: 'Description 2',
    disabled: true,
  },
]

const getArgs = <T, >(
  options: CheckboxGroupOption<T>[],
  defaultValue: T[],
  disabled = false,
  onChange?: (changeEvent: T[]) => void
): CheckboxGroupProps<T> => ({
    options,
    defaultValue,
    isDisabled: disabled,
    onChange,
  })

const meta = {
  component: CheckboxGroup<number>,
  args: getArgs<number>(baseOptions, [1]),
} satisfies Meta<typeof CheckboxGroup<number>>

export default meta

type Story = StoryObj<typeof meta>;

export const BasicExample: Story = {}

export const WithDescription: Story = {
  args: getArgs<number>(optionsWithDescription, [2]),
}

export const Disabled: Story = {
  args: getArgs(baseOptions, [1], true),
}

export const PartiallyDisabled: Story = {
  args: getArgs(optionsPartiallyDisabled, [1]),
}

export const AllOptionsDisabled: Story = {
  args: getArgs(allOptionsDisabled, [1]),
}

const onChange: (changeEvent: number[]) => void = (event) => alert(`selected value is now ${event}`)
export const WithOnChange: Story = {
  args: getArgs(baseOptions, [1], false, onChange),
}

type MoreComplexType = { a: number; b: number };

const objectOption1: MoreComplexType = { a: 1, b: 2 }
const objectOption2: MoreComplexType = { a: 3, b: 4 }

const props: CheckboxGroupProps<MoreComplexType> = {
  options: [
    { value: objectOption1, label: 'option 1' },
    { value: objectOption2, label: 'option 2' },
  ],
  defaultValue: [objectOption2],
  isDisabled: false,
}

const metaO = {
  component: CheckboxGroup<MoreComplexType>,
  args: props,
} satisfies Meta<typeof CheckboxGroup<MoreComplexType>>

type StoryO = StoryObj<typeof metaO>;

export const WithObjectValueType: StoryO = {
  args: props,
}
