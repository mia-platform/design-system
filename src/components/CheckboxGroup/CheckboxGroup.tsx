/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, useCallback, useMemo, useState } from 'react'
import { Checkbox as AntCheckbox } from 'antd'

import { Checkbox } from '../Checkbox'
import { CheckboxGroupProps } from './props.ts'
import styles from './CheckboxGroup.module.css'

export const CheckboxGroup = <T, >(
  {
    defaultValue,
    isDisabled,
    options,
    onChange,
  }: CheckboxGroupProps<T>
): ReactElement => {
  const [value, setValue] = useState(defaultValue)

  const checkboxOptions = useMemo(() => {
    return options?.map((option, index) => {
      return (
        <Checkbox
          description={option.description}
          isDisabled={option.disabled || isDisabled}
          /* eslint-disable-next-line react/no-array-index-key */
          key={index}
          label={option.label}
          value={option.value}
        />
      )
    })
  }, [isDisabled, options])

  const handleChange = useCallback((val: T[]) => {
    setValue(val)
    if (onChange) {
      onChange(val)
    }
  }, [onChange])

  return (
    <AntCheckbox.Group
      value={value}
      onChange={handleChange}
    >
      <div className={styles.options}>
        {checkboxOptions}
      </div>
    </AntCheckbox.Group>
  )
}
