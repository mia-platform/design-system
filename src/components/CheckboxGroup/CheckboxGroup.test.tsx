/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { render, screen, waitFor, within } from '@testing-library/react'

import { CheckboxGroupOption, CheckboxGroupProps } from './props'
import { CheckboxGroup } from './CheckboxGroup'
import { userEvent } from '../../test-utils.tsx'

const baseOptions: CheckboxGroupOption<number>[] = [
  {
    value: 1,
    label: 'checkbox 1',
  },
  {
    value: 2,
    label: 'checkbox 2',
  },
]

const optionsWithDescription: CheckboxGroupOption<number>[] = [
  {
    ...baseOptions[0],
    description: 'description 1',
  },
  {
    ...baseOptions[0],
    description: 'description 2',
  },
]

const partiallyDisabledOptions: CheckboxGroupOption<number>[] = [
  {
    ...baseOptions[0],
    description: 'description 1',
    disabled: true,
  },
  {
    ...baseOptions[0],
    description: 'description 2',
  },
]

const baseProps: CheckboxGroupProps<number> = {
  options: baseOptions,
}

describe('CheckboxGroup', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('renders correctly', () => {
    it('with base props', () => {
      const { asFragment } = render(<CheckboxGroup {...baseProps} />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('with descriptions', () => {
      const { asFragment } = render(
        <CheckboxGroup {...{ ...baseProps, options: optionsWithDescription }} />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('when disabled', () => {
      const { asFragment } = render(
        <CheckboxGroup {...{ ...baseProps, isDisabled: true }} />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('with partially disabled options', () => {
      const { asFragment } = render(
        <CheckboxGroup {...{ ...baseProps, options: partiallyDisabledOptions }} />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('horizontal', () => {
      const { asFragment } = render(
        <CheckboxGroup {...{ ...baseProps, direction: CheckboxGroup.Direction.Horizontal }} />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('should invoke onChange with correct value on user selection', async() => {
    const onChange = jest.fn()
    const props: CheckboxGroupProps<number> = {
      ...baseProps,
      onChange,
    }

    render(<CheckboxGroup {...props} />)

    const firstCheckbox = within(screen.getByText('checkbox 1')).getByRole('checkbox')
    const secondCheckbox = within(screen.getByText('checkbox 2')).getByRole('checkbox')

    expect(firstCheckbox).not.toBeChecked()
    expect(secondCheckbox).not.toBeChecked()

    await userEvent.click(firstCheckbox)
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith([baseOptions[0].value])
    })
    expect(firstCheckbox).toBeChecked()

    await userEvent.click(secondCheckbox)
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith([
        baseOptions[0].value,
        baseOptions[1].value,
      ])
    })

    expect(secondCheckbox).toBeChecked()
  })

  it('should invoke onChange with correct value on user deselection', async() => {
    const onChange = jest.fn()
    const props: CheckboxGroupProps<number> = {
      ...baseProps,
      defaultValue: [baseOptions[0].value],
      onChange,
    }

    render(<CheckboxGroup {...props} />)

    const checkbox = within(screen.getByText('checkbox 1')).getByRole('checkbox')

    expect(checkbox).toBeChecked()

    await userEvent.click(checkbox)
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith([])
    })
    expect(checkbox).not.toBeChecked()
  })

  it('should not trigger onChange if disabled', async() => {
    const onChange = jest.fn()
    const props: CheckboxGroupProps<number> = {
      ...baseProps,
      isDisabled: true,
      onChange,
    }

    render(<CheckboxGroup {...props} />)

    const checkbox = within(screen.getByText('checkbox 1')).getByRole('checkbox')

    await waitFor(() => {
      expect(userEvent.click(checkbox)).rejects.toBeTruthy()
    })
  })

  it('should work fine with controlled values', async() => {
    const onChange = jest.fn()
    const props: CheckboxGroupProps<number> = {
      ...baseProps,
      value: [baseOptions[0].value],
      onChange,
    }

    render(<CheckboxGroup {...props} />)

    const firstCheckbox = within(screen.getByText('checkbox 1')).getByRole('checkbox')
    const secondCheckbox = within(screen.getByText('checkbox 2')).getByRole('checkbox')

    expect(firstCheckbox).toBeChecked()
    expect(secondCheckbox).not.toBeChecked()

    await userEvent.click(firstCheckbox)
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith([])
    })
    expect(firstCheckbox).toBeChecked()

    await userEvent.click(secondCheckbox)
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith([baseOptions[1].value])
    })

  })

})
