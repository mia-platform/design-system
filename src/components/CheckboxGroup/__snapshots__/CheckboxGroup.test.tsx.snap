// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup renders correctly horizontal 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group horizontal css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="options"
    >
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
          </div>
        </span>
      </label>
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="2"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 2
            </div>
          </div>
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckboxGroup renders correctly when disabled 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="options"
    >
      <label
        class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9 ant-checkbox-disabled"
        >
          <input
            class="ant-checkbox-input"
            disabled=""
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
          </div>
        </span>
      </label>
      <label
        class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9 ant-checkbox-disabled"
        >
          <input
            class="ant-checkbox-input"
            disabled=""
            type="checkbox"
            value="2"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 2
            </div>
          </div>
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckboxGroup renders correctly with base props 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="options"
    >
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
          </div>
        </span>
      </label>
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="2"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 2
            </div>
          </div>
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckboxGroup renders correctly with descriptions 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="options"
    >
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              description 1
            </div>
          </div>
        </span>
      </label>
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              description 2
            </div>
          </div>
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckboxGroup renders correctly with partially disabled options 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-5wsri9"
  >
    <div
      class="options"
    >
      <label
        class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9 ant-checkbox-disabled"
        >
          <input
            class="ant-checkbox-input"
            disabled=""
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              description 1
            </div>
          </div>
        </span>
      </label>
      <label
        class="ant-checkbox-wrapper checkbox css-dev-only-do-not-override-5wsri9"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-5wsri9"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
            value="1"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          <div
            class="checkboxContent"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              checkbox 1
            </div>
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              description 2
            </div>
          </div>
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
